src/app/store/accounts/blockchain/bitcoin/bitcoin-keychain.ts

```diff
@@ -97,3 +106,56 @@ export function bitcoinSignerFactory<T extends BitcoinSignerFactoryArgs>(args: T
     };
   };
 }
+
+interface CreateSignersForAllNetworkTypesArgs {
+  mainnetKeychainFn: (accountIndex: number) => HDKey;
+  testnetKeychainFn: (accountIndex: number) => HDKey;
+  paymentFn: (keychain: HDKey, network: BitcoinNetworkModes) => unknown;
+}
+function createSignersForAllNetworkTypes<T extends CreateSignersForAllNetworkTypesArgs>({
+  mainnetKeychainFn,
+  testnetKeychainFn,
+  paymentFn,
+}: T) {
+  return ({ accountIndex, addressIndex }: { accountIndex: number; addressIndex: number }) => {
+    const mainnetAccount = mainnetKeychainFn(accountIndex);
+    const testnetAccount = testnetKeychainFn(accountIndex);
+
+    function makeNetworkSigner(keychain: HDKey, network: BitcoinNetworkModes) {
```

Comment:
Extracted how, as in outside of the fn?

Made this really just to avoid some repetition, and also not using a type cast to make sure the `payment` is the correct type.

Reply:
Merging to avoid further conflicting changes, but open to other ideas about how to make this logic clearer if not